openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -177,10 +179,11 @@ def set_user_preference(requesting_user, preference_key, preference_value, usern
     existing_user = _get_authorized_user(requesting_user, username)
     serializer = create_user_preference_serializer(existing_user, preference_key, preference_value)
     validate_user_preference_serializer(serializer, preference_key, preference_value)
-    try:
-        serializer.save()
-    except Exception as error:
-        raise _create_preference_update_error(preference_key, preference_value, error)
+    if serializer.instance is None or serializer.instance.value != serializer.validated_data['value']:
+        try:
+            serializer.save()
+        except Exception as error:
```

Comment:
Sorry.  I scanned too quickly.  The reraise as another error actually makes me happier with this.